src/cursor/cur_layered.c

```diff
@@ -1508,6 +1509,24 @@ __clayered_insert(WT_CURSOR *cursor)
     F_CLR(cursor, WT_CURSTD_KEY_SET | WT_CURSTD_VALUE_SET);
 
     WT_STAT_CONN_DSRC_INCR(session, layered_curs_insert);
+
+    if (0) {
```

Comment:
I would strongly prefer this was moved to a different function instead of a goto + `if (0)` block.

Reply:
how would that look like? You still need a jump label to skip all the previous logic to execute this logic.

Reply:
What about moving to here
```
if (ret == 0)
            goto duplicate;
```

Reply:
I don't think you do, because the current code will fall through to the existing `goto err`, which is the same as what happens with the new label when you reach the end of the `if (0)` block.

Reply:
I don't follow